File: src/main/java/tn/esprit/nafseyti/models/JournalEntry.java
package tn.esprit.nafseyti.models;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

public class JournalEntry {
    private int id;
    private int userId;
    private java.sql.Date date;
    private String humeur; // "excellent", "bien", "moyen", "difficile", "très_difficile"
    private List<String> emotions; // Liste d'émotions
    private String noteTexte;
    private int energie; // 1-10
    private int sommeilQualite; // 1-10
    private Timestamp createdAt;

    // Constructeurs
    public JournalEntry() {
        this.emotions = new ArrayList<>();
    }

    public JournalEntry(int userId, java.sql.Date date, String humeur, String noteTexte, int energie, int sommeilQualite) {
        this.userId = userId;
        this.date = date;
        this.humeur = humeur;
        this.noteTexte = noteTexte;
        this.energie = energie;
        this.sommeilQualite = sommeilQualite;
        this.emotions = new ArrayList<>();
    }

    // Getters et Setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public java.sql.Date getDate() {
        return date;
    }

    public void setDate(java.sql.Date date) {
        this.date = date;
    }

    public String getHumeur() {
        return humeur;
    }

    public void setHumeur(String humeur) {
        this.humeur = humeur;
    }

    public List<String> getEmotions() {
        return emotions;
    }

    public void setEmotions(List<String> emotions) {
        this.emotions = emotions;
    }

    public void addEmotion(String emotion) {
        if (!this.emotions.contains(emotion)) {
            this.emotions.add(emotion);
        }
    }

    public String getEmotionsAsString() {
        return String.join(",", emotions);
    }

    public void setEmotionsFromString(String emotionsStr) {
        this.emotions.clear();
        if (emotionsStr != null && !emotionsStr.isEmpty()) {
            String[] emotionArray = emotionsStr.split(",");
            for (String emotion : emotionArray) {
                this.emotions.add(emotion.trim());
            }
        }
    }

    public String getNoteTexte() {
        return noteTexte;
    }

    public void setNoteTexte(String noteTexte) {
        this.noteTexte = noteTexte;
    }

    public int getEnergie() {
        return energie;
    }

    public void setEnergie(int energie) {
        this.energie = Math.max(1, Math.min(10, energie)); // Limiter entre 1 et 10
    }

    public int getSommeilQualite() {
        return sommeilQualite;
    }

    public void setSommeilQualite(int sommeilQualite) {
        this.sommeilQualite = Math.max(1, Math.min(10, sommeilQualite)); // Limiter entre 1 et 10
    }

    public Timestamp getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Timestamp createdAt) {
        this.createdAt = createdAt;
    }

    // Méthodes utilitaires
    public String getHumeurEmoji() {
        switch (humeur) {
            case "excellent": return "😊";
            case "bien": return "🙂";
            case "moyen": return "😐";
            case "difficile": return "😔";
            case "très_difficile": return "😰";
            default: return "😐";
        }
    }

    public int getHumeurScore() {
        switch (humeur) {
            case "excellent": return 5;
            case "bien": return 4;
            case "moyen": return 3;
            case "difficile": return 2;
            case "très_difficile": return 1;
            default: return 3;
        }
    }

    @Override
    public String toString() {
        return "JournalEntry{" +
                "id=" + id +
                ", userId=" + userId +
                ", date=" + date +
                ", humeur='" + humeur + '\'' +
                ", emotions=" + emotions +
                ", energie=" + energie +
                ", sommeil=" + sommeilQualite +
                '}';
    }
}

File: src/Entity/JournalEntry.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\JournalEntryRepository;

#[ORM\Entity(repositoryClass: JournalEntryRepository::class)]
#[ORM\Table(name: 'journal_entries')]
class JournalEntry
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\ManyToOne(targetEntity: User::class, inversedBy: 'journalEntries')]
    #[ORM\JoinColumn(name: 'user_id', referencedColumnName: 'id', nullable: false)]
    private ?User $user = null;

    #[ORM\Column(type: 'date')]
    private ?\DateTimeInterface $date = null;

    #[ORM\Column(type: 'string')]
    private ?string $humeur = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $emotions = null;

    #[ORM\Column(name: 'note_texte', type: 'text', nullable: true)]
    private ?string $noteTexte = null;

    #[ORM\Column(type: 'integer')]
    private ?int $energie = null;

    #[ORM\Column(name: 'sommeil_qualite', type: 'integer')]
    private ?int $sommeilQualite = null;

    #[ORM\Column(name: 'created_at', type: 'datetime')]
    private ?\DateTimeInterface $createdAt = null;

    #[ORM\Column(name: 'updated_at', type: 'datetime')]
    private ?\DateTimeInterface $updatedAt = null;

    public function __construct()
    {
        $this->createdAt = new \DateTime();
        $this->updatedAt = new \DateTime();
    }

    public function getId(): ?int { return $this->id; }

    public function getUser(): ?User { return $this->user; }
    public function setUser(?User $user): self { $this->user = $user; return $this; }

    public function getDate(): ?\DateTimeInterface { return $this->date; }
    public function setDate(\DateTimeInterface $date): self { $this->date = $date; return $this; }

    public function getHumeur(): ?string { return $this->humeur; }
    public function setHumeur(string $humeur): self { $this->humeur = $humeur; return $this; }

    public function getEmotions(): ?string { return $this->emotions; }
    public function setEmotions(?string $emotions): self { $this->emotions = $emotions; return $this; }

    public function getNoteTexte(): ?string { return $this->noteTexte; }
    public function setNoteTexte(?string $noteTexte): self { $this->noteTexte = $noteTexte; return $this; }

    public function getEnergie(): ?int { return $this->energie; }
    public function setEnergie(int $energie): self { $this->energie = $energie; return $this; }

    public function getSommeilQualite(): ?int { return $this->sommeilQualite; }
    public function setSommeilQualite(int $sommeilQualite): self { $this->sommeilQualite = $sommeilQualite; return $this; }

    public function getCreatedAt(): ?\DateTimeInterface { return $this->createdAt; }
    public function setCreatedAt(\DateTimeInterface $createdAt): self { $this->createdAt = $createdAt; return $this; }

    public function getUpdatedAt(): ?\DateTimeInterface { return $this->updatedAt; }
    public function setUpdatedAt(\DateTimeInterface $updatedAt): self { $this->updatedAt = $updatedAt; return $this; }
}